### PR TITLE
Add test for S_OBJNAME & update test for LF_BUILDINFO cl and cmd

### PR DESCRIPTION
**Pull request by nebulark**
Update the unit test for checking cl and cmd in LF_BUILDINFO. With llvm-pdbutil we can now more specifically check if the string appears at the right location instead of just checking whether the string exists at all.
Context: https://github.com/rust-lang/rust/issues/96475

### CODE REVIEW

File: tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs [hunk 1]
```diff
@@ -17,23 +17,9 @@ fn main() {
...
-}
 
-fn assert_pdb_contains(needle: &str) {
-    let needle = needle.as_bytes();
-    use bstr::ByteSlice;
-    assert!(&rfs::read("my_crate_name.pdb").find(needle).is_some());
+    let pdbutil_result =
+        llvm::LlvmPdbutil::new().arg("dump").arg("-ids").input("my_crate_name.pdb").run();
+
+    llvm::llvm_filecheck().patterns("filecheck.txt").stdin_buf(pdbutil_result.stdout_utf8()).run();
```
Comment (jieyouxu) [inline tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs:24]: Question: is there a specific reason why this needs to use filecheck?

File: tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs [hunk 2]
```diff
@@ -17,23 +17,9 @@ fn main() {
...
-        assert_pdb_contains(test);
-    }
-}
 
-fn assert_pdb_contains(needle: &str) {
-    let needle = needle.as_bytes();
-    use bstr::ByteSlice;
-    assert!(&rfs::read("my_crate_name.pdb").find(needle).is_some());
+    let pdbutil_result =
+        llvm::LlvmPdbutil::new().arg("dump").arg("-ids").input("my_crate_name.pdb").run();
```
Comment (jieyouxu) [inline tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs:22]: Suggestion: can you add a simple helper like for the other llvm bin utils that just do

```rs
pub fn llvm_pdbutil() -> LlvmPdbutil {
    LlvmPdbutil::new()
}
```

for consistency?

Comment (nebulark) [reply tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs:22]: Sure. Will add this in the other pr (https://github.com/rust-lang/rust/pull/130155) and then fix the pr that did not get merged first.

File: tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs (hunk 1 above)
Comment (nebulark) [reply tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs:24]: I'm not aware/have't looked for alternatives. The order those strings appear is important.
What would you recommend as alternative?
Is using filecheck discouraged in general?

File: tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs (hunk 2 above)
Comment (nebulark) [reply tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs]: Hmm. Maybe its easier if I just close https://github.com/rust-lang/rust/pull/130155 and add its changes to this pr. Then we don't have do deal with merge problems.
Is this fine for you?

Comment (jieyouxu) [reply tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs]: Yes that's fine if you want to collapse that PR into this one.

File: tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs (hunk 1 above)
Comment (jieyouxu) [reply tests/run-make/pdb-buildinfo-cl-cmd/rmake.rs:24]: Right, ordering is important. In that case, I think this is fine.

**Comment by jieyouxu** [comment]
r=me after collapsing the other PR into this one.
@rustbot author

**Review by jieyouxu — APPROVED** [review]
Thanks!

**Pull request merged**